File: SDK/WebClient/src/APIDefinitions/IRuntimeApi.ts
import { ApiDefinition, methodStub } from '@remote-ioc/runtime';
import { AccountInfo, TypedEventEmitter } from '../Types';

@ApiDefinition('runtime-api')
export class IRuntimeApi extends TypedEventEmitter<{ updated: [string] }> {
  public async getConnectedAccounts(): Promise<AccountInfo[]> {
    methodStub(this);
  }

  public async getLastBattleResponse(accountId: string): Promise<unknown> {
    methodStub(this, accountId);
  }
}

File: src/Shared/DataModel/APIs/IRuntimeApi.cs
using System;
using System.Threading.Tasks;
using Newtonsoft.Json.Linq;

namespace Raid.Toolkit.DataModel
{
    [PublicApi("runtime-api")]
    public interface IRuntimeApi
    {
        [PublicApi("updated")]
        event EventHandler<SerializableEventArgs> Updated;

        [PublicApi("getConnectedAccounts")]
        Task<Account[]> GetConnectedAccounts();

        [PublicApi("getLastBattleResponse")]
        Task<JObject> GetLastBattleResponse(string accountId);
    }
}
